table.cc

```diff
@@ -480,6 +474,48 @@ table::make_reader(schema_ptr s,
     }
 }
 
+flat_mutation_reader
+table::make_reader(schema_ptr s,
```

Comment:
We have a method `table::as_mutation_source()` which calls this `table::make_reader()` to make its reader, which now in turn makes another internal `mutation_source` which it optionally restricts.
Perhaps it would be cleaner to have an private version of `as_mutation_source_internal()`, which wraps `table::do_make_reader()` and have all other method create their `mutation_source`:s with this.
The public `as_mutation_source()` would look like this:
```
auto ms = as_mutation_source_internal();
if (needs_restricting()) {
    return restrict_mutation_source(ms);
}
return ms;
```
Just a suggestion.

Reply:
Yes, I can call it as_unrestricted_mutation_source().